src/Soil/rre.jl

```diff
@@ -214,3 +214,137 @@ function ClimaLSM.make_update_aux(model::RichardsModel)
     end
     return update_aux!
 end
+
+
+"""
+"""
+struct RichardsTridiagonalW{R, J, W, T} <: ClimaLSM.AbstractTridiagonalW
+    # reference to dtγ, which is specified by the ODE solver
+    dtγ_ref::R
+    # derivative of implicit tendency - used to fill Jacobian
+    ∂ϑₜ∂ϑ::J
+    # array of tridiagonal matrices containing W for each column
+    W_column_arrays::W
+    # caches used to evaluate ldiv!
+    temp1::T
+    temp2::T
+    # whether this struct is used to compute Wfact_t or Wfact
+    transform::Bool
+end
+
+
+function RichardsTridiagonalW(
+    Y::ClimaCore.Fields.FieldVector,
+    FT,
+    space,
+    transform::Bool,
+)
+    N = Spaces.nlevels(space)
+
+    tridiag_type = Operators.StencilCoefs{-1, 1, NTuple{3, FT}}
+    ∂ϑₜ∂ϑ = Fields.Field(tridiag_type, space)
+
+    ∂ϑₜ∂ϑ.coefs[1] .= FT(0)
+    ∂ϑₜ∂ϑ.coefs[2] .= FT(0)
+    ∂ϑₜ∂ϑ.coefs[3] .= FT(0)
+
+    arr1 = Array{FT}(undef, N - 1)
+
+    W_column_arrays = [
+        LinearAlgebra.Tridiagonal(
+            zeros(FT, N - 1) .+ FT(0),
+            zeros(FT, N) .+ FT(0),
+            zeros(FT, N - 1) .+ FT(0),
+        ) for _ in 1:Threads.nthreads()
+    ]
+    dtγ_ref = Ref(FT(0))
+    #Not initialized!!
+    temp1 = similar(Y.soil.ϑ_l)
+    temp2 = similar(Y.soil.ϑ_l)
+
+    return RichardsTridiagonalW(
+        dtγ_ref,
+        ∂ϑₜ∂ϑ,
+        W_column_arrays,
+        temp1,
+        temp2,
+        transform,
+    )
+end
+
+
+"""
+    """
```

Comment:
We should add a docstring explaining that `update_jacobian!` should be used as `Wfact!` for CTS/ODE.

Reply:
will do!